packages/hathor-rpc-handler/src/rpcMethods/getBalance.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { HathorWallet } from '@hathor/wallet-lib';
+import { GetBalanceObject } from '@hathor/wallet-lib/lib/wallet/types';
+import {
+  ConfirmationPromptTypes,
+  GetBalanceConfirmationPrompt,
+  GetBalanceRpcRequest,
+  PromptHandler,
+} from '../types';
+import { NotImplementedError, PromptRejectedError } from '../errors';
+import { validateNetwork } from '../helpers';
+
+/**
+ * Gets the balance for specified tokens using the provided wallet.
+ *
+ * @param rpcRequest - The RPC request containing the parameters for getting the balance.
+ * @param wallet - The wallet instance to use for retrieving the balance.
+ * @param promptHandler - A function to handle prompts for user confirmation.
+ *
+ * @returns The balances of the specified tokens.
+ *
+ * @throws {NotImplementedError} - If address indexes are specified, which is not implemented.
+ * @throws {PromptRejectedError} - If the user rejects the balance confirmation prompt.
+ */
+export async function getBalance(
+  rpcRequest: GetBalanceRpcRequest,
+  wallet: HathorWallet,
+  promptHandler: PromptHandler,
+) {
+  const { network, tokens, addressIndexes } = rpcRequest.params;
+
+  if (addressIndexes) {
+    throw new NotImplementedError();
+  }
+
+  validateNetwork(wallet, network);
+
+  const balances: GetBalanceObject[] = await Promise.all(
+    tokens.map(token => wallet.getBalance(token)),
+  );
+
+  const prompt: GetBalanceConfirmationPrompt = {
+    type: ConfirmationPromptTypes.GetBalanceConfirmationPrompt,
```

Comment:
Why do you need to confirm this with prompt? Don't you need just to return it?

Reply:
No, we don't want the dApp to be able to get balances without user approval